### PR TITLE
[9.x] Vite

### PR DESCRIPTION
**Pull request by jessarcher**
This PR includes documentation for Vite as the default asset compilation tool.

### CODE REVIEW

**Comment by jessarcher** [comment]
> I don't think we should delete the Mix docs here. All back linking will be broken. Instead, just remove it from the navigation.

The page hasn't been deleted, just trimmed down a lot and moved in the navigation.

**Comment by driesvints** [comment]
@jessarcher ah I see. No worries then!

File: vite.md [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+npm run build
+```
+
+<a name="working-with-scripts"></a>
+## Working With JavaScript
+
+<a name="aliases"></a>
+### Aliases
+
+The Laravel plugin provides a common aliases to help you hit the ground running:
```
Comment (tontonsb) [inline vite.md:162]: aliases -> alias

File: vite.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+Within a fresh installation of Laravel, you will find a `package.json` file in the root of your application's directory structure. The default `package.json` file already includes everything you need to get started using Vite and the Laravel plugin. You may install your application's frontend dependencies by running:
+
+```sh
+npm install
+```
+
+<a name="configuring-vite"></a>
+### Configuring Vite
+
+Vite is configured in a `vite.config.js` file in the root of your project. You are free to customize this file based on your needs and you may also install any other plugins your application requires, such as `@vitejs/vue-plugin` or `@vitejs/react-plugin`.
```
Comment (mpociot) [inline vite.md:71]: The plugins are `@vitejs/plugin-vue` and `@vitejs/plugin-react`

File: vite.md (hunk 1 above)
Comment (timacdonald) [reply vite.md]: Fixed. Thank you!

File: vite.md (hunk 2 above)
Comment (timacdonald) [reply vite.md]: Fixed. Thank you!

File: frontend.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+         * @param  int  $id
+         * @return \Inertia\Response
+         */
+        public function show($id)
+        {
+            return Inertia::render('Users/Profile', [
+                'user' => User::findOrFail($id)
+            ]);
+        }
+    }
```
Comment (jessarcher) [inline frontend.md:141]:
```suggestion
```php
<?php

namespace App\Http\Controllers;

use App\Http\Controllers\Controller;
use App\Models\User;
use Inertia\Inertia;

class UserController extends Controller
{
    /**
     * Show the profile for a given user.
     *
     * @param  int  $id
     * @return \Inertia\Response
     */
    public function show($id)
    {
        return Inertia::render('Users/Profile', [
            'user' => User::findOrFail($id)
        ]);
    }
}
```
```

File: frontend.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+<a name="livewire"></a>
+### Livewire
+
+[Laravel Livewire](https://laravel-livewire.com) is a framework for building Laravel powered frontends that feel dynamic, modern, and alive just like frontends built with modern JavaScript frameworks like Vue and React.
+
+When using Livewire, you will create Livewire "components" that render a discrete portion of your UI and exposes methods and data that can be invoked and interacted with from your application's frontend. For example, a simple "Counter" component might look like the following:
+
+```php
+use Livewire\Component;
```
Comment (jessarcher) [inline frontend.md:65]: For consistency with the Inertia example below, do you think we should add this?

```suggestion
<?php

namespace App\Http\Livewire;

use Livewire\Component;
```

**Pull request merged**